upcloud/client/client.go

```diff
@@ -33,7 +40,18 @@ type Client struct {
 
 // New creates ands returns a new client configured with the specified user and password
 func New(userName, password string) *Client {
-	return NewWithHTTPClient(userName, password, cleanhttp.DefaultClient())
+	httpClient := &http.Client{}
+	httpTransport := cleanhttp.DefaultTransport()
+
+	if upcloud.IsEnvEnabled(APIUnsecureSkipVerifyEnv) {
```

Comment:
Do note that NewWithHTTPClient should also support the same semantics. 

Although, I'm not sure how it would / should work if UPCLOUD_API_INSECURE_SKIP_VERIFY = true and the custom client does not include that..